### PR TITLE
[14.0][MIG] mail

### PR DESCRIPTION
**Pull request by RaulOForgeFlow**
--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: openupgrade_scripts/scripts/mail/14.0.1.0/pre-migration.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    openupgrade.rename_models(
+        env.cr, [("email_template.preview", "mail.template.preview")]
+    )
+    openupgrade.set_xml_ids_noupdate_value(env, "mail", "mail_channel_rule", True)
```
Comment (MiquelRForgeFlow) [inline openupgrade_scripts/scripts/mail/14.0.1.0/pre-migration.py]:
```suggestion
    openupgrade.set_xml_ids_noupdate_value(env, "mail", ["mail_channel_rule"], True)
```

**Review by MiquelRForgeFlow — APPROVED** [review]

**Comment by pedrobaeza** [comment]
@MiquelRForgeFlow is the CI already working?

**Comment by MiquelRForgeFlow** [comment]
No

**Comment by pedrobaeza** [comment]
Then we can't merge this without having such safeguard.

File: openupgrade_scripts/scripts/mail/14.0.1.0/pre-migration.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+        env.cr, [("email_template.preview", "mail.template.preview")]
+    )
+    openupgrade.set_xml_ids_noupdate_value(env, "mail", ["mail_channel_rule"], True)
+    openupgrade.update_module_moved_fields(
+        env.cr, "mail.mail", ["description"], "website_mail", "mail"
+    )
+    openupgrade.update_module_moved_fields(
+        env.cr, "mail.message", ["description"], "website_mail", "mail"
+    )
+    openupgrade.rename_columns(env.cr, [("field", None)])
```
Comment (jcdrubay) [inline openupgrade_scripts/scripts/mail/14.0.1.0/pre-migration.py]: This triggers an error:

`AttributeError: 'list' object has no attribute 'keys'`

Comment (jcdrubay) [reply openupgrade_scripts/scripts/mail/14.0.1.0/pre-migration.py]: I think you want to do this: `openupgrade.copy_columns(env.cr, {"mail_tracking_value": [("field", None, None)]})`

File: openupgrade_scripts/scripts/mail/14.0.1.0/post-migration.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+
+def fill_mail_tracking_value_field(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE mail_tracking_value mtv
+        SET field = imf.id
+        FROM ir_model_fields imf
+        JOIN mail_message mm ON mtv.mail_message_id = mm.id
```
Comment (jcdrubay) [inline openupgrade_scripts/scripts/mail/14.0.1.0/post-migration.py]:
```suggestion
        JOIN mail_message mm ON imf.model = mm.model
        WHERE imf.name = mtv.{} AND mtv.mail_message_id = mm.id
```

File: openupgrade_scripts/scripts/mail/14.0.1.0/post-migration.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+            openupgrade.get_legacy_name("field")
+        ),
+    )
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    fill_mail_tracking_value_field(env)
+    openupgrade.load_data(
+        env.cr, "mail", "openupgrade_scripts/scripts/mail/14.0.1.0/noupdate_changes.xml"
```
Comment (jcdrubay) [inline openupgrade_scripts/scripts/mail/14.0.1.0/post-migration.py]:
```suggestion
        env.cr, "mail", "14.0.1.0/noupdate_changes.xml"
```

File: openupgrade_scripts/scripts/mail/14.0.1.0/pre-migration.py (hunk 2 above)
Comment (MiquelRForgeFlow) [reply openupgrade_scripts/scripts/mail/14.0.1.0/pre-migration.py]:
```suggestion
    openupgrade.rename_columns(env.cr, {"mail_tracking_value": [("field", None)]})
```

**Comment by pedrobaeza** [comment]
@RaulOForgeFlow please check the changes I have added in the subsequent comment for refining the scripts, but very good introduction. Merging now.

**Pull request merged**